src/util/make-knex.js

```diff
@@ -44,11 +44,9 @@ module.exports = function makeKnex(client) {
             batch = _.chunk(batch, chunkSize)
           }
 
-          Promise.all(batch.map((items) => {
+          return Promise.all(batch.map((items) => {
             return tr(table).insert(items)
-          }))
-          .then(tr.commit)
-          .catch(tr.rollback)
+          }));
```

Comment:
Why was the commit/rollback lines removed here? Since upgrading the version `0.95.11`, all `batchInsert` calls are hanging and I think this may be why

Reply:
Good catch, I'll take a look.

Reply:
@jshjohnson Having checked the code and the tests, I believe that code behaves correctly. Autocommit/autorollback happens when no explicit transaction is passed, and when explicit transaction is passed, knex should not be committing automatically, because user might want to perform some additional actions within the same transaction. Hence commit or rollback need to be invoked explicitly on that transaction by the user.